### PR TITLE
ref(tags): Refactors and updates tag summary components to have better accessibility support

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Refactors and updates tag summary components to have better accessibility support
- updates components to use `<ol>` and `<li>`
- adds more and better accessibility labels
- updates expand/collapse toggle component to be button

### CODE REVIEW

File: static/app/components/group/tagFacets/tagFacetsDistributionMeter.tsx [hunk 1]
```diff
@@ -152,32 +159,40 @@ function TagFacetsDistributionMeter({
...
-            <Link
-              key={`segment-${segment.name}-${index}`}
-              to={segment.url}
-              aria-label={t(
-                'Add the %s %s segment tag to the search query',
+          const isOtherSegment =
+            index === topSegments.length - 1 && segment.value === 'other';
+          const linkLabel = isOtherSegment
+            ? t(
+                'Other %s tag values, %s of all events. View all tags.',
```
Comment (shruthilayaj) [inline static/app/components/group/tagFacets/tagFacetsDistributionMeter.tsx:171]: "other" label isn't clickable in issue details (looks like the link is an empty string)
also in all cases, clicking on these labels in the issue details page take you to the tag details page so this aria label is misleading

File: static/app/components/group/tagFacets/tagFacetsDistributionMeter.tsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import styled from '@emotion/styled';
 import debounce from 'lodash/debounce';
```
Comment (shruthilayaj) [inline static/app/components/group/tagFacets/tagFacetsDistributionMeter.tsx:4]: segmentProps for <SegmentBar> in this file could also use aria-labels (I see we have it for "Other" but not the other segments)
Screen Shot 2023-01-25 at 3 54 40 PM

**Review by shruthilayaj — APPROVED** [review]

File: static/app/components/group/tagFacets/tagFacetsDistributionMeter.tsx (hunk 2 above)
Comment (edwardgou-sentry) [reply static/app/components/group/tagFacets/tagFacetsDistributionMeter.tsx:4]: Added labels for segments!

File: static/app/components/group/tagFacets/tagFacetsDistributionMeter.tsx (hunk 1 above)
Comment (edwardgou-sentry) [reply static/app/components/group/tagFacets/tagFacetsDistributionMeter.tsx:177]: 👍 Handling this in an upcoming pr!

**Pull request merged**